ingestion/src/metadata/data_insight/producer/entity_producer.py

```diff
@@ -0,0 +1,69 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Producer class for data insight entity reports
+"""
+
+import traceback
+from typing import Iterable
+
+from metadata.data_insight.producer.producer_interface import ProducerInterface
+from metadata.generated.schema.entity.data import (
+    chart,
+    container,
+    dashboard,
+    database,
+    databaseSchema,
+    mlmodel,
+    pipeline,
+    table,
+    topic,
+)
+from metadata.utils.logger import data_insight_logger
+
+logger = data_insight_logger()
+
+
+class EntityProducer(ProducerInterface):
+    """entity producer class"""
+
+    entities = [
+        chart.Chart,
+        dashboard.Dashboard,
+        database.Database,
+        databaseSchema.DatabaseSchema,
+        mlmodel.MlModel,
+        pipeline.Pipeline,
+        table.Table,
+        topic.Topic,
+        container.Container,
+    ]
+
+    # pylint: disable=protected-access,dangerous-default-value
+    def fetch_data(self, limit=100, fields=["*"]) -> Iterable:
```

Comment:
to not have the `dangerous-default-value` we could do a `fields: Optional[List[str]] = None`, and then:

```python
yield from self.metadata.list_all_entities(
    entity, limit=limit, fields=fields or ["*"]
)
```

Reply:
It is kind of a false flag here. We don't do operations on the we just pass it as a parameter to the endpoint. The problem with what you suggest is that you can't determine if you pass None, to have no fields (maybe you don't care about the fields or None so that you return all the fields.